### ISSUE TITLE
feat: add a monorepo-wide `npm update`

### ISSUE THREAD

**Issue by mesqueeb** [JamieMason/syncpack#32]
## Description

Hi, in regular `npm` projects, i often use `npm update` which does:
1. update the `package.json` file: change the package versions to the highest possible
2. actually update all the packages in node_modules based on this

Inside `lerna` projects, this is sadly not possible, because `npm` cannot install stuff in subfolders.
And afaik there's no good equivalent to `npm update` in lerna:
https://github.com/lerna/lerna/issues/2142
(issue been open for ages)

## Suggested Solution

if syncpack could offer to do step 1 above:
1. update the `package.json` file: change the package versions to the highest possible

then afterwards I could run `lerna bootstrap` to actually install those.

Keep up the good work!!

**Comment by JamieMason** [comment]
Thanks @mesqueeb, this is a good idea and I may add this as a feature in the future. It would be low priority because there are other tools which do this, but it would fit well and could be a nice addition.

Incase it's useful to you, the way I do this today is to use [npm-check-updates](https://www.npmjs.com/package/npm-check-updates) to update the versions in package.json files, then run a regular install afterwards.

1. Run `ncu -u` to update the root package.json.
1. Run `lerna exec 'ncu -u'` to update each package.json in the monorepo.

If lerna is installed locally to your project instead of as a global dependency you can run this instead... 

If you use yarn:

```
yarn lerna exec 'ncu -u'
```

Or this if you use npm:

```
node_modules/.bin/lerna exec 'ncu -u'
```

**Comment by JamieMason** [comment]
I've decided not to do this as there are a lot of good tools out there for updating npm dependencies, I don't want to dilute the focus/purpose of syncpack too much. Thanks anyway for raising this.

**Comment by JamieMason** [comment]
Reopening as work has actually started on this, things are a bit different now and I think this could be a good fit. 
Please subscribe to this issue to track progress.

Thanks @mesqueeb.

/cc @cimchd @hgenru (👍🏻'd this issue)

**Comment by JamieMason** [comment]
Released in [10.7.2](https://github.com/JamieMason/syncpack/releases/tag/10.7.2)
